Added madctl parameter to ILI9341 driver for hardware rotation and RGB/BGR control, enabling faster rendering and more flexibility

ritesh006 opened the pull request:
### Summary
This PR adds support for passing a `madctl` parameter to the `ILI9341` driver, allowing developers to directly control hardware rotation via the MADCTL register.  
It also preserves support for BGR/RGB selection and inversion.

### Motivation
Currently, display rotation relies on displayio’s software-based handling, which can be inefficient.  
By exposing MADCTL, developers can take advantage of the LCD controller’s built-in rotation logic, which improves performance (especially for OnDiskBitmaps).

### Changes
- Added an optional `madctl` parameter to the `ILI9341` constructor.
- Retained existing `bgr` and `invert` options.
- Updated initialization sequence handling to correctly append MADCTL.
- Updated README usage examples to include `madctl`.

### Tests
- Added unit tests (`test_madctl.py`) covering:
  - Default RGB encoding
  - BGR encoding
  - Custom MADCTL overrides
- All tests pass locally.

### Related Issues
Closes #26 ("Make it easy to set MADCTL for rotation").

ritesh006 committed:
ili9341: add optional madctl kwarg, remove hardcoded writes

ritesh006 commented:
Please review this commit

ladyada commented:
reviewed. we will not merge, please use/add/adapt kwargs instead of hardcoding MADCTL writes

ritesh006 commented:
Hi @ladyada your review comment addressed please take a look.

ladyada commented:
hi, to be clear: we will not accept passing in MADCTL. instead, you can add kwargs for any/each setting that may be required (e.g. color order)

ritesh006 committed:
ili9341: add color_order kwarg, keep bgr compat, update tests/docs

ritesh006 commented:
Hi @ladyada I’ve updated this PR as suggested: replaced raw madctl with a color_order kwarg (default "RGB", optional "BGR"), kept bgr for compatibility, and updated docs/tests. Ready for re-review.

FoamyGuy commented on adafruit_ili9341.py:
```diff
@@ -91,16 +99,48 @@ class ILI9341(BusDisplay):
     """
     ILI9341 display driver
 
-    :param FourWire bus: bus that the display is connected to
```
The FourWire bus argument should not get removed from the docstrings I think

FoamyGuy commented on adafruit_ili9341.py:
```diff
@@ -91,16 +99,48 @@ class ILI9341(BusDisplay):
+        *,
+        width=240,
+        height=320,
+        rotation=0,
```
Please add docstring listings for the new args `width`, `height`, and `rotation` all arguments should be listed in the docstrings above.

FoamyGuy commented on pyproject.toml:
```diff
@@ -46,3 +46,16 @@ py-modules = ["adafruit_ili9341"]
 dependencies = {file = ["requirements.txt"]}
 optional-dependencies = {optional = {file = ["optional_requirements.txt"]}}
+
+[tool.ruff]
```
We don't generally want to make configuration changes to pyproject.toml in single repos like this. 

We have `ruff.toml` file with relatively configuration inside of it that is used for all of the library repos.

If there are configuration changes that we should consider then we'll want to discuss doing so more broadly with a patch across all repos instead of having it changed one-off in PRs like this.

Please revert the changes in pyproject.toml for now.
